PayItForward.ConsoleClient.UnitTests/Web/Controllers/StoriesController/Donate_Should.cs

```diff
@@ -0,0 +1,251 @@
+            this.storiesService = new Mock<IStoriesService>();
+            this.donationService = new Mock<IDonationsService>();
+            this.usersService = new Mock<IUsersService>();
+            this.httpaccessor = new Mock<IHttpContextAccessor>();
+            this.storiesController = new PayItForward.Web.Controllers.StoriesController(
+                this.storiesService.Object,
+                this.donationService.Object,
+                this.usersService.Object,
+                this.mapper.Object,
+                this.httpaccessor.Object);
+        }
+
+        [Fact]
+        public void ReturnARedirectToIndexHomeWhenIdIsEmpty()
+        {
+            // Arrange
+
+            // Act
+            var result = this.storiesController.Donate(id: Guid.Empty);
+
+            // Assert
+            var redirectToActionResult = Assert.IsType<RedirectToActionResult>(result);
+            Assert.Equal("Stories", redirectToActionResult.ControllerName);
+            Assert.Equal("Details", redirectToActionResult.ActionName);
+        }
+
+        [Fact]
+        public void ReturnContent_WithStoryNotFound_WhenStoryNotFound()
+        {
+            // Arrange
+            this.storiesService.Setup(s => s.GetStoryById(this.storyId))
+                .Returns((StoryDTO)null);
+
+            // Act
+            var result = this.storiesController.Donate(this.storyId);
+
+            // Assert
+            var contentResult = Assert.IsType<ContentResult>(result);
+            Assert.Equal("Story not found.", contentResult.Content);
+        }
+
+        [Fact]
+
+        public void ReturnResultContent_WithNoUserToDonate_WhenDonatorIdIsNullOrEmpty()
+        {
+            // Arrange
+            this.storiesService.Setup(s => s.GetStoryById(this.storyId))
+                .Returns(this.HelperStoryDto().FirstOrDefault());
+            this.storiesController.ControllerContext = new ControllerContext
+            {
+                HttpContext = new DefaultHttpContext
+                {
+                    User = new ClaimsPrincipal(
+                    new ClaimsIdentity(
+                        new Claim[]
+                        {
+                            new Claim(ClaimTypes.NameIdentifier, "username")
+                        },
+                        "someAuthTypeName"))
+                }
+            };
+            this.usersService.Setup(u => u.GetUserById(" ")).Returns((UserDTO)null);
+
+            // Act
+            var result = this.storiesController.Donate(this.HelperDonateViewModel(), this.storyId);
+
+            // Assert
+            var contentResult = Assert.IsType<ContentResult>(result);
+            Assert.Equal("No user to donate.", contentResult.Content);
+        }
+
+        [Fact]
+        public void ReturnContentResult_WithStoryNotFound_WhenGivenInvalidModel()
+        {
+            // Arrange & Act
+            var result = this.storiesController.Donate(model: null, id: this.storyId);
```

Comment:
The test you test does not expect Story not found when model is null, but when GetStoryId returns null, please change this test